[IMP] accounting: add new version of doc about taxes (#703)
odoo/documentation-user: accounting/fiscality/taxes/taxes.rst

=====
Taxes
=====

> There are numerous types of **taxes**, and their application varies greatly, depending mostly on
> your company's localization.
> 
> It is crucial to record all taxes with accuracy. Therefore, Odoo's tax engine supports all kinds of
> uses and computations.

Reviewer: ```suggestion
There are numerous types of **taxes**, and their application varies greatly, depending mostly on
your company's localization. And to make sure they are recorded with accuracy, Odoo's tax engine supports all kinds of
uses and computations.
```

.. _taxes/list_activation:

Activate Sales Taxes from the List view
=======================================

As part of your :doc:`Fiscal Localization Package
<../../fiscal_localizations/overview/fiscal_localization_packages>`, most of your country's sales
taxes are already preconfigured on your database. However, only a few of them are activated by
default, so that you can activate only the ones relevant for your business.

To activate Sale Taxes, go to :menuselection:`Accounting --> Configuration --> Taxes` and use the
*Activate* toggle button to activate or deactivate a tax.

.. image:: media/taxes-list.png
   :align: center
   :alt: Activate pre-configured taxes in Odoo Accounting

.. _taxes/configuration:

Configuration
=============

To edit or create a **Tax**, go to :menuselection:`Accounting --> Configuration --> Taxes` and open
a tax or click on *Create*.

.. image:: media/taxes-edit.png
   :align: center
   :alt: Edition of a tax in Odoo Accounting

Basic Options
-------------

> 

Author: I think the ref `.. _taxes/name` wouldn't work properly this way.
Rule of thumb: a white line separates different commands, I think. That's what I remember, at least...

  For example, a product has a Sales Price of $1000, and we apply a *10% of Price* tax. We then
  have:

  +-------------+-------------+----------+----------+
  | Product's   | Price       | Tax      | Total    |
  | Sales Price | without tax |          |          |
  +=============+=============+==========+==========+
  | 1,000       | 1,000       | 100      | 1,100.00 |
  +-------------+-------------+----------+----------+

- **Percentage of Price Tax Included**

  The *Total* is the taxable basis: the tax's amount is a percentage of the Total.

  For example, a product has a Sales Price of $1000, and we apply a *10% of Price Tax Included* tax.
  We then have:

  +-------------+-------------+----------+----------+
  | Product's   | Price       | Tax      | Total    |
  | Sales Price | without tax |          |          |
  +=============+=============+==========+==========+
  | 1,000       | 1,000       | 111.11   | 1,111.11 |
  +-------------+-------------+----------+----------+

.. _taxes/active:

Active
~~~~~~

Only **active** taxes can be added to new documents.

.. important::
   It is not possible to delete taxes that have been used to record an entry. Instead, you can

>    deactivate these taxes to prevent users from using them.

Author: Much better indeed, thanks!

.. _taxes/definition-tab:

Definition tab
--------------

Allocate with precision the amount of the taxable basis or percentages of the computed tax to
multiple accounts and Tax Grids.

.. image:: media/taxes-definition.png
   :align: center
   :alt: Allocate tax amounts to the right accounts and tax grids

- **Based On**:

  - Base: The price on the invoice line
  - % of tax: a percentage of the computed tax.

- **Account**: if defined, an additional Journal Item is recorded.
- **Tax Grids**: Tax Grids are used to generate  your :doc:`Tax Reports
  <../../reporting/declarations/tax_returns>` automatically, according to your country's regulations.

.. _taxes/advanced-tab:

Advanced Options tab
--------------------

.. _taxes/label-invoices:

Label on Invoices
~~~~~~~~~~~~~~~~~

The label of the tax, as displayed on each invoice line in the **Taxes** column.

.. image:: media/taxes-invoice-label.png
   :align: center
   :alt: The Label on Invoices is displayed on each invoice line

.. important::
   The label displayed between the Subtotal and Total lines is defined by the :ref:`Tax Group

>    <taxes/tax-group>` name and *not* by the Label on Invoices.

Author: No it's good you're pointing this out.
But in the section above I write this about Tax Names:

```
Tax Name
~~~~~~~~

The **Tax Name** as you want to display it on Odoo.

.. note::
   To modify how the tax is labeled on invoices, please see below: :ref:`taxes/label-invoices`.
```

So your test might have had the same tax name and label, but the tax name is only in backend :)

>    <taxes/tax-group>` name and *not* by the Label on Invoices.

Author: And indeed, the way it works IS confusing. It tool me some while to figure it out correctly...
The kind of things that is _really_ time-consuming ^^

>    <taxes/tax-group>` name and *not* by the Label on Invoices.

Author: @kbr-odoo 

I'm adding this in the "Configuration" section, with a reference to it in the other sections. What do you think?

.. _taxes/appellations:

```
.. important::
   Taxes have three different appellations, altough each one is linked to a different field with a 
   specific use. Refer to the following table to see where they are displayed.

   +----------------+-------------------------+----------------------+
   | :ref:`Tax Name | :ref:`Label on Invoice  | :ref:`Tax Group      |
   | <taxes/name>`  | <taxes/label-invoices>` | <taxes/tax-group>`   |
   +================+=========================+======================+
   | Backend        | *Taxes* column          | Above the *Total*    |
   |                | on invoices             | line on invoices     |
   +----------------+-------------------------+----------------------+

```

>    <taxes/tax-group>` name and *not* by the Label on Invoices.

Reviewer: But I did try with different names, and the name I set on *Label on Invoice* was not the one used. I am doing something wrong then: https://nimb.ws/Lm3aBW.

I believe the doc might have too many ..notes and ..important sections about it now. Once it is explained under the title, like it was before, it might be enough. Otherwise, the doc also becomes longer, which might give the impression of difficulty, something we want to avoid.
:)

suggestion: appelations -> names

>    <taxes/tax-group>` name and *not* by the Label on Invoices.

Author: @kbr-odoo 
No worries :)
But you are still in back end.
I you click on preview, or print the invoice, or see it on the customer portal, it is the Label on Invoice that is displayed.
The tax name is only for you as a back end user.

>    <taxes/tax-group>` name and *not* by the Label on Invoices.

Reviewer: got it.. thank you, Jon ^^

>    <taxes/tax-group>` name and *not* by the Label on Invoices.

Reviewer: suggestions to consider:

- Not have this information repeated many times (keep it like it was in your first version). Maybe adding the word *Customer* before *Invoice* + more contextualized screenshots.
- suggestion: appelations -> names

o/o/o/

>    <taxes/tax-group>` name and *not* by the Label on Invoices.

Author: Last commit with all your suggested recommendations (although I choose "labels" instead of "names" to avoid confusion in the texts).

   following documentation: :doc:`B2B_B2C`.

.. note::
   - **Invoices**: By default, the Line Subtotals displayed on your invoices are *Tax-Excluded*. To
     display *Tax-Included* Line Subtotals, go to :menuselection:`Accounting --> Configuration -->
     Settings --> Customer Invoices`, and select *Tax-Included* in the **Line Subtotals Tax
     Display** field, then click on *Save*.
   - **eCommerce**: By default, the prices displayed on your eCommerce website are *Tax-Excluded*.
     To display *Tax-Included* prices, go to :menuselection:`Website --> Configuration --> Settings
     --> Pricing`, and select *Tax-Included* in the **Product Prices** field, then click on *Save*.

.. _taxes/base-subsequent:

Affect Base of Subsequent Taxes
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

With this option, the total tax-included becomes the taxable basis for the other taxes applied to
the same product.

You can configure a new :ref:`Group of Taxes <taxes/computation>` to include this tax, or add it
directly to a product line.

.. image:: media/taxes-subsequent-line.png
   :align: center
   :alt: The eco-tax is taken into the basis of the 21% VAT tax

.. warning::

>    The order in which you add the taxes on a product line has no consequence on how the amounts are
>    computed. If you add these taxes directly on a product line, only the tax sequence determines the
>    order in which the taxes are applied. You can reorder the sequence by going to
>    :menuselection:`Accounting --> Configuration --> Taxes` and moving the lines in the list with
>    the handles next to the tax names.

Reviewer: ```suggestion
   The order in which you add the taxes on a product line has no effect on how amounts are
   computed. If you add taxes directly on a product line, only the tax sequence determines the
   order in which they are applied. To reorder the sequence, go to
   :menuselection:`Accounting --> Configuration --> Taxes`, and drag and drop lines.
```

To activate Sale Taxes, go to :menuselection:`Accounting --> Configuration --> Taxes` and use the
*Activate* toggle button to activate or deactivate a tax.

.. image:: media/taxes-list.png
   :align: center
   :alt: Activate pre-configured taxes in Odoo Accounting

.. _taxes/configuration:

Configuration
=============

To edit or create a **Tax**, go to :menuselection:`Accounting --> Configuration --> Taxes` and open
a tax or click on *Create*.

.. image:: media/taxes-edit.png
   :align: center
   :alt: Edition of a tax in Odoo Accounting

.. _taxes/labels:

.. important::
   Taxes have three different labels, each one having a specific use. Refer to the following table to
   see where they are displayed.

   +------------------+-------------------------+-------------------------+
   | :ref:`Tax Name   | :ref:`Label on Invoice  | :ref:`Tax Group         |
   | <taxes/name>`    | <taxes/label-invoices>` | <taxes/tax-group>`      |
   +==================+=========================+=========================+
   | Back end         | *Taxes* column on       | Above the *Total* line  |
   |                  | exported invoices       | on exported invoices    |

>    +------------------+-------------------------+-------------------------+

Reviewer 2: Joli :+1:

Author: Merci @AntoineVDV   😄

- **Percentage of Price**

  The *Sales Price* is the taxable basis: the tax's amount is computed by multiplying the Sales
  Price by the tax's percentage.

  For example, a product has a Sales Price of $1000, and we apply a *10% of Price* tax. We then
  have:

  +-------------+-------------+----------+----------+
  | Product's   | Price       | Tax      | Total    |
  | Sales Price | without tax |          |          |
  +=============+=============+==========+==========+
  | 1,000       | 1,000       | 100      | 1,100.00 |
  +-------------+-------------+----------+----------+

- **Percentage of Price Tax Included**

  The *Total* is the taxable basis: the tax's amount is a percentage of the Total.

  For example, a product has a Sales Price of $1000, and we apply a *10% of Price Tax Included* tax.
  We then have:

  +-------------+-------------+----------+----------+
  | Product's   | Price       | Tax      | Total    |
  | Sales Price | without tax |          |          |
  +=============+=============+==========+==========+
  | 1,000       | 1,000       | 111.11   | 1,111.11 |
  +-------------+-------------+----------+----------+

.. _taxes/active:

Active
~~~~~~

> Only **active** taxes can be added to new documents.

Reviewer: ```suggestion
Only **Active** taxes can be added to new documents.
```

.. image:: media/taxes-invoice-label.png
   :align: center
   :alt: The Label on Invoices is displayed on each invoice line

.. important::
   The label displayed between the Subtotal and Total lines is defined by the :ref:`Tax Group
   <taxes/tax-group>` name and *not* by the Label on Invoices.

.. note::
   This is *not* the label displayed between the Subtotal and Total lines. Refer to :ref:`this table
   <taxes/appellations>` for more information.

.. _taxes/tax-group:

Tax Group
~~~~~~~~~

Select to which **Tax Group** the tax belongs. The Tax Group name is the label displayed above the
*Total* line on invoices.

.. image:: media/taxes-invoice-tax-group.png
   :align: center
   :alt: The Tax Group name is different from the Label on Invoices

Tax groups include different iterations of the same tax. This can be useful when you must record
differently the same tax according to :doc:`Fiscal Positions <fiscal_positions>`.

For the example above, we see a 0% tax for Intra-Community customers in Europe. It records amounts
on specific accounts and with specific tax grids. Still, to the customer, it is a 0% tax. That's why

> the :ref:`Label on the Invoice <taxes/label-invoices>` indicates *0% EU*, but the Tax Group name,

Reviewer: ```suggestion
the :ref:`Label on the Invoice <taxes/label-invoices>` indicates *0% EU*, AND (?) the Tax Group name,
```
